client/src/components/overview/ProductOverview.jsx

```diff
@@ -42,35 +45,65 @@ class ProductOverview extends React.Component {
   }
 
   loadAdditionalProductData(id) {
+    console.log('Loading Product Overview data...');
     if (!id || id === null) {
       console.log('Product ID is null');
       return;
     }
+
     if (this.props.isTesting) {
-      this.ratings = this.props.ratings;
-      this.updateRatingsProperties(this.ratings);
-      this.styles = this.props.styles;
-      this.updateDefaultStyle();
+      this.updateRatingsProperties(this.props.ratings);
+      this.updateDefaultStyle(this.props.styles);
       return;
     }
-    Promise.all([this.fetchRatings(id), this.fetchProductStyles(id)])
-      .then(res => {
-        this.ratings = res[0].data.ratings;
-        this.updateRatingsProperties(this.ratings);
-        this.styles = res[1].data.results;
-        this.updateDefaultStyle();
-      })
-      .catch(error => {
-        console.error(error.stack);
-      });
+
+    this.fetchRatings(id, this.updateRatingsProperties);
+    this.fetchProductStyles(id, (styles) => {
+      this.setState({ styles });
```

Comment:
This is unlikely to work the way you want component state to work. If you keep updating the same `styles` object, React will fail to notice and nothing will update. If you are careful to create a new `styles` object every time you set state, React will re-render everything that uses this data, even if some/all of the actual content is the same.